mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala

```diff
@@ -287,7 +282,7 @@ abstract class GeneralizedLinearAlgorithm[M <: GeneralizedLinearModel]
         if (useFeatureScaling) {
           input.map(lp => (lp.label, scaler.transform(lp.features))).cache()
         } else {
-          input.map(lp => (lp.label, lp.features))
+          input.map(lp => (lp.label, lp.features)).cache()
```

Comment:
That makes sense too. I suppose changing behavior less is good. Yes, then the safer thing is to just unpersist only if the `data` RDD is persisted.